internal/pkg/queue/persistedqueue.go

```diff
@@ -0,0 +1,34 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package queue
+
+import "github.com/elastic/elastic-agent/internal/pkg/fleetapi"
+
+// PersistedQueue is an action queue with a Save function for persistency.
+type PersistedQueue struct {
+	ActionQueue
+	qs persistor
+}
+
+type persistor interface {
+	SetQueue(a []fleetapi.Action)
+	Save() error
+}
+
+// NewPersistedQueue creates a persisted queue from an existing action queue and persistor.
+//
+// The persistor the minimal interface needed from the state storeage mechanism.
+func NewPersistedQueue(q *ActionQueue, qs persistor) *PersistedQueue {
```

Comment:
Is there ever a case where the `ActionQueue` is used without it now being a `PersistedQueue`? If not, then it would be better to just merge the `ActionQueue` and `PersistedQueue` and make the `ActionQueue` a persisted queue.

Adding layer on top of another layer just makes things confusing and in v2 the goal was to remove so many layers of abstraction. If one queue is only every used by the Elastic Agent then lets just have one queue.

Reply:
good point, i'll change it